src/floyd-warshall.jl

```diff
@@ -36,17 +36,17 @@ end
 #     Note that it is possible to consume large amounts of memory as the
 #     space required for the FloydWarshallState is O(n^2).
 #     """ ->
-function floyd_warshall_shortest_paths(
+function floyd_warshall_shortest_paths{T<:Real}(
     g::AbstractGraph;
-    edge_dists::AbstractArray{Float64, 2} = Array(Float64,(0,0))
+    edge_dists::AbstractArray{T,2} = zeros(0,0)
 )
 
     use_dists = issparse(edge_dists)? nnz(edge_dists > 0) : !isempty(edge_dists)
     n_v = nv(g)
-    dists = fill(convert(Float64,Inf), (n_v,n_v))
-    parents = zeros(Int, (n_v,n_v))
+    S = typeof(one(T) + one(T))
+    dists = fill(typemax(S), n_v, n_v)
```

Comment:
> Keep in mind that this will be the first type parameterization in LightGraphs.

Well, it turns out that this statement is false. There are already polymorphic constructors for [Graph](https://github.com/JuliaGraphs/LightGraphs.jl/blob/14b1ef56ebc5be16bda8f5f5b24d9b1f05e61f21/src/graph.jl#L32-45) and [Digraph](https://github.com/JuliaGraphs/LightGraphs.jl/blob/14b1ef56ebc5be16bda8f5f5b24d9b1f05e61f21/src/digraph.jl#L29-42)...

Furthermore, the type parameter probably should not be `T<:Number`, but rather `T<:Real` or even `T<:Integer`. The entries of adjacency matrices are counts of edges (possibly signed due to directedness), and I do not know of any nonintegral or nonreal generalizations of the notion of adjacency matrix.